treeherder/changelog/collector.py

```diff
@@ -0,0 +1,96 @@
+""" Collector, grabs changes in various sources and put them in a DB.
+"""
+import github3
+import json
+import os
+
+from treeherder.changelog import models
+from treeherder.changelog.filters import filter_out
+
+
+MAX_ITEMS = 100
+CFG = os.path.join(os.path.dirname(__file__), "repositories.json")
+
+with open(CFG) as f:
+    CFG = json.loads(f.read())
+
+
+class GitHub:
+    def __init__(self):
+        # XXX plug into Django-whatever-secrets-things
+        self.token = os.environ["GITHUB_TOKEN"]
+        self.id = os.environ["GITHUB_ID"]
+        self.gh = github3.login(token=self.token, two_factor_callback=self._2FA)
+
+    def _2FA(self):
+        code = ""
+        while not code:
+            # The user could accidentally press Enter before being ready,
+            # let's protect them from doing that.
+            code = input("Enter 2FA code: ")
```

Comment:
Why is the 2FA callback needed?

Reply:
When it's added to the heroku scheduler, is there some other token/value we'd need to use instead?

Reply:
@sarah-clements  I don't know. I guess we need to see what Github account is used. if it's not 2fa enabled then we won't call that _2fa function. We can add a specific raise in deployment to avoid calling input()

Reply:
I don't have a confirmation on this. Let's try setting it up to skip the _2fa function except for use with localhost. Then I'll deploy this to prototype to test the cron job on Monday or Tuesday. I need to know what the command and any flags will be for use in production mode.

Reply:
ok great, I'll add that.

you can just run the "update_changelog" command with --days=7, that should start to feed some data in